tests/datastreams/test_public_api.py

```diff
@@ -30,3 +30,19 @@ def test_public_api():
                 sorted(["direction:in", "manual_checkpoint:true", "type:kinesis", "topic:stream-123"]), parent_hash
             )
             assert got.hash == expected
+
+
+def test_manual_checkpoint_behavior():
```

Comment:
nit: this is testing: `automatic checkpoints` behavior. Let's rename that test function.